### PR TITLE
Refactor Credential Signing

### PR DESCRIPTION
**Pull request by nitro-neal**
This change slightly refactors credential signing and verifying. 

Instead of `PortableDid` we now use `BearerDid`

```javascript
import { DidKey } from '@web5/dids';
const issuer: BearerDid = await DidKey.create();

const vcJwt = await vc.sign({ did: issuer });
```

This change also:
* Changes jwt signing to use bearer did's `signer` function 
* Updates all unit tests to use `BearerDid`
* Updates to the following versions
    "@web5/common": "0.2.3",
    "@web5/crypto": "0.4.0",
    "@web5/dids": "0.4.0"

### CODE REVIEW

File: packages/credentials/package.json [hunk 1]
```diff
@@ -75,9 +75,9 @@
   },
   "dependencies": {
     "@sphereon/pex": "2.1.0",
-    "@web5/common": "0.2.2",
-    "@web5/crypto": "0.2.4",
-    "@web5/dids": "0.2.4"
+    "@web5/common": "0.2.3",
```
Comment (nitro-neal) [inline packages/credentials/package.json:78]: These are the latest versions right?

**Comment by nitro-neal** [comment]
Should I bump the version of the credentials package to "version": "0.4.2", ?

File: packages/credentials/package.json (hunk 1 above)
Comment (frankhinek) [reply packages/credentials/package.json:78]: Yes -- `0.4.0` for both dids and crypto -- `0.2.3` for common.

File: packages/credentials/src/jwt.ts [hunk 2]
```diff
@@ -189,18 +146,11 @@ export class Jwt {
...
-    const algorithmId = `${decodedJwt.header.alg}:${publicKeyJwk['crv'] || ''}`;
-    if (!(algorithmId in Jwt.algorithms)) {
-      throw new Error(`Verification failed: ${algorithmId} not supported`);
-    }
-
-    const { signer, options: signatureAlgorithm } = Jwt.algorithms[algorithmId];
-
-    const isSignatureValid = await signer.verify({
-      algorithm : signatureAlgorithm!,
+    const crypto = new CryptoApi();
```
Comment (nitro-neal) [inline packages/credentials/src/jwt.ts:149]: Should we just have is initialized once at the top?

File: packages/credentials/src/jwt.ts [hunk 3]
```diff
@@ -136,14 +99,8 @@ export class Jwt {
     const toSign = `${base64UrlEncodedHeader}.${base64UrlEncodedPayload}`;
     const toSignBytes = Convert.string(toSign).toUint8Array();
 
-    const algorithmId = `${header.alg}:${privateKeyJwk['crv'] || ''}`;
-    if (!(algorithmId in Jwt.algorithms)) {
-      throw new Error(`Signing failed: ${algorithmId} not supported`);
-    }
-
-    const { signer, options: signatureAlgorithm } = Jwt.algorithms[algorithmId];
+    const signatureBytes = await signer.sign({data: toSignBytes});
```
Comment (frankhinek) [inline packages/credentials/src/jwt.ts:102]:
```suggestion
    const signatureBytes = await signer.sign({ data: toSignBytes });
```

File: packages/credentials/src/jwt.ts (hunk 2 above)
Comment (frankhinek) [reply packages/credentials/src/jwt.ts:149]: sounds good

**Review by frankhinek — APPROVED** [review]
Looks good other than a few changes noted above and updating the test

**Review by mistermoe — APPROVED** [review]
cc: @angiejones

**Comment by frankhinek** [comment]
> Should I bump the version of the credentials package to "version": "0.4.2", ?

Yes, we might as well publish a release today so that `tbdex` repos can use it.

**Pull request merged**